### PR TITLE
native drivers test

### PR DESCRIPTION
**Pull request by juimonen**
test native drivers build

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>

### CODE REVIEW

**Comment by lgirdwood** [comment]
@juimonen @abonislawski good work guys - I was able to move dai-legacy.h and dai-zephyr.h to the xtos/zephyr folders without too much pain. @juimonen can you pls complete my topic/lrg/rtos-dai branch for the dai header move. It shows we still use some xtos headers for dai api, but the number is getting smaller :)

**Comment by lgirdwood** [comment]
@juimonen how is this looking and the dai.h move looking. Do we have an ETA ?

**Comment by juimonen** [comment]
> @juimonen how is this looking and the dai.h move looking. Do we have an ETA ?

This has now some experimental commits for the hda dai -> but if you look at my last comments in https://github.com/zephyrproject-rtos/zephyr/pull/51000, we probably need to go first with the 19 hda nodes.
I tried to spend last couple of days thinking/testing different options to make the hda dt "clean", but concerning all parties 
requirements -> we should just stick with the model in the pr, just fix the warnings.

So test results are good for all dais -> we should/could move to native immediately after above hda dai pr is merged (with tgl DT entries).

File: src/lib/dai.c [hunk 1]
```diff
@@ -157,8 +158,22 @@ static const struct device *dai_get_zephyr_device(uint32_t type, uint32_t index)
 
 	for (i = 0; i < ARRAY_SIZE(zephyr_dev); i++) {
 		cfg = dai_config_get(zephyr_dev[i], dir);
-		if (cfg->type == type && cfg->dai_index == index)
-			return zephyr_dev[i];
+		switch (cfg->type) {
```
Comment (lyakh) [inline src/lib/dai.c:161]: could simplify this a bit by placing
```
if (cfg->type != type)
	continue;
```
before the `switch`

**Comment by juimonen** [comment]
closing, everything merged in other PRs.

**Pull request closed without merging**